# --- T2-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# 
# T2 SDE: package/.../zsnes/gcc43.patch
# Copyright (C) 2009 The T2 SDE Project
# 
# More information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- T2-COPYRIGHT-NOTE-END ---

--- src/tools/depbuild.cpp.vanilla	2009-02-16 11:06:47.000000000 +0100
+++ src/tools/depbuild.cpp	2009-02-16 11:07:22.000000000 +0100
@@ -24,6 +24,8 @@
 #include <iostream>
 #include <string>
 #include <cstdio>
+#include <stdlib.h>
+
 using namespace std;
 
 #include "fileutil.h"
@@ -183,7 +185,7 @@
   }
 }
 
-int main(size_t argc, const char *const *const argv)
+int main(int argc, const char *const *const argv)
 {
   if (argc < 5)
   {
--- src/tools/strutil.h.vanilla	2009-02-16 11:06:29.000000000 +0100
+++ src/tools/strutil.h	2009-02-16 11:06:37.000000000 +0100
@@ -22,6 +22,7 @@
 #ifndef STRUTIL_H
 #define STRUTIL_H
 
+#include <string.h>
 #include <string>
 #include <vector>
 #include <cctype>
--- src/parsegen.cpp.vanilla	2009-02-16 11:09:11.000000000 +0100
+++ src/parsegen.cpp	2009-02-16 11:09:35.000000000 +0100
@@ -33,6 +33,7 @@
 #include <stack>
 using namespace std;
 
+#include <string.h>
 #include <errno.h>
 #include <zlib.h>
 
@@ -1828,7 +1829,7 @@
   }
 }
 
-int main(size_t argc, const char **argv)
+int main(int argc, const char **argv)
 {
   const char *cheader_file = 0;
   bool compile = false;
